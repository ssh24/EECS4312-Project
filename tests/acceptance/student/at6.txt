-- this mainly tests some incorrect inputs and error regarding mistyping pid when
-- creating a new container

-- create the tracker
new_tracker(4, 2) 

-- create two phases p1 and p2
new_phase("p1", "packing", 4, <<glass, metal,metal, plastic, liquid>>) 
new_phase("p2", "crushing", 3, <<glass, metal, plastic>>)
new_phase("p3", "crushing", 3, <<metal,metal,metal,metal,metal>>)
new_phase("p4", "crushing", 3, <<glass,glass, plastic, liquid>>)

-- add several containers 
new_container("c1", [glass,   1.2], " p1")
new_container("c1", [glass,   1.2], "@p1")
new_container("c1", [glass,   1.2], "p12")
new_container("c1", [glass,   1.2], "p1")
new_container("c2", [plastic,   1.5], "p1")
new_container("c3", [metal,   0.5], "p2")




