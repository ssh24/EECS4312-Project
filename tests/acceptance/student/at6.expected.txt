  state 0 ok
  max_phase_radiation: 0.00, max_container_radiation: 0.00
  phases: pid->name:capacity,count,radiation
  containers: cid->pid->material,radioactivity
->new_tracker(4,2)
  state 1 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
  containers: cid->pid->material,radioactivity
->new_phase("p1","packing",4,<<glass, metal, metal, plastic, liquid>>)
  state 2 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,0,0.00,{glass,metal,plastic,liquid}
  containers: cid->pid->material,radioactivity
->new_phase("p2","crushing",3,<<glass, metal, plastic>>)
  state 3 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,0,0.00,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->new_phase("p3","crushing",3,<<metal, metal, metal, metal, metal>>)
  state 4 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,0,0.00,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
    p3->crushing:3,0,0.00,{metal}
  containers: cid->pid->material,radioactivity
->new_phase("p4","crushing",3,<<glass, glass, plastic, liquid>>)
  state 5 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,0,0.00,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
    p3->crushing:3,0,0.00,{metal}
    p4->crushing:3,0,0.00,{glass,plastic,liquid}
  containers: cid->pid->material,radioactivity
->new_container("c1",[glass, 1.2]," p1")
  state 6 e5: identifiers/names must start with A-Z, a-z or 0..9
->new_container("c1",[glass, 1.2],"@p1")
  state 7 e5: identifiers/names must start with A-Z, a-z or 0..9
->new_container("c1",[glass, 1.2],"p12")
  state 8 e9: phase identifier not in the system
->new_container("c1",[glass, 1.2],"p1")
  state 9 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,1,1.20,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
    p3->crushing:3,0,0.00,{metal}
    p4->crushing:3,0,0.00,{glass,plastic,liquid}
  containers: cid->pid->material,radioactivity
    c1->p1->glass,1.20
->new_container("c2",[plastic, 1.5],"p1")
  state 10 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,2,2.70,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
    p3->crushing:3,0,0.00,{metal}
    p4->crushing:3,0,0.00,{glass,plastic,liquid}
  containers: cid->pid->material,radioactivity
    c1->p1->glass,1.20
    c2->p1->plastic,1.50
->new_container("c3",[metal, 0.5],"p2")
  state 11 ok
  max_phase_radiation: 4.00, max_container_radiation: 2.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,2,2.70,{glass,metal,plastic,liquid}
    p2->crushing:3,1,0.50,{glass,metal,plastic}
    p3->crushing:3,0,0.00,{metal}
    p4->crushing:3,0,0.00,{glass,plastic,liquid}
  containers: cid->pid->material,radioactivity
    c1->p1->glass,1.20
    c2->p1->plastic,1.50
    c3->p2->metal,0.50
